internal/security/secretstore/init.go

```diff
@@ -275,3 +327,96 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 	lc.Info("proxy certificate pair are uploaded to secret store successfully, Vault init done successfully")
 	return false
 }
+
+func makeTokenIssuingToken(
+	lc logger.LoggingClient,
+	configuration *config.ConfigurationStruct,
+	tokenMaintenance *TokenMaintenance,
+	fileOpener fileioperformer.FileIoPerformer,
+	rootToken string) (RevokeFunc, error) {
+
+	configAdminTokenPath := configuration.SecretService.TokenProviderAdminTokenPath
+
+	// Create delegate credential for use by the token provider
+	tokenIssuingToken, revokeIssuingTokenFuc, err := tokenMaintenance.CreateTokenIssuingToken(rootToken)
+	if err != nil {
+		lc.Error(fmt.Sprintf("failed to create token issuing token %s", err.Error()))
+		return nil, err
+	}
+	lc.Info("created token issuing token")
+
+	// Write the token issuing token to disk to pass it to the token provider
+	if configAdminTokenPath == "" {
+		lc.Error(fmt.Sprintf("TokenProviderAdminTokenPath is a required configuration setting"))
+		revokeIssuingTokenFuc()
+		return nil, err
+	}
+	adminTokenPath, err := filepath.Abs(configAdminTokenPath)
+	if err != nil {
+		lc.Error(fmt.Sprintf("failed to convert to absolute path %s: %s", configAdminTokenPath, err.Error()))
+		revokeIssuingTokenFuc()
+		return nil, err
+	}
+	dirOfAdminToken := filepath.Dir(adminTokenPath)
+	err = fileOpener.MkdirAll(dirOfAdminToken, 0700)
+	if err != nil {
+		lc.Error(fmt.Sprintf("failed to create tokenpath base dir: %s", err.Error()))
+		revokeIssuingTokenFuc()
+		return nil, err
+	}
+	tokenWriter, err := fileOpener.OpenFileWriter(adminTokenPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+	if err != nil {
+		lc.Error(fmt.Sprintf("failed to create token issuing file %s: %s", adminTokenPath, err.Error()))
+		revokeIssuingTokenFuc()
+		return nil, err
+	}
+
+	encoder := json.NewEncoder(tokenWriter)
+	if encoder == nil {
+		lc.Error(fmt.Sprintf("failed to create token encoder: %s", err.Error()))
```

Comment:
`err` is nil in this case...  you may want to create a new err here and return that err.

Reply:
Good catch!

Reply:
Fixed.